roachtest: kv0/enc=false/nodes=3/cpu=32/size=64kb failed

roachtest.kv0/enc=false/nodes=3/cpu=32/size=64kb [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2940327&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2940327&tab=artifacts#/kv0/enc=false/nodes=3/cpu=32/size=64kb) on master @ [d84625c42d11e38775b6a1ff3f4089e839fe3e82](https://github.com/cockroachdb/cockroach/commits/d84625c42d11e38775b6a1ff3f4089e839fe3e82):


```
		  |   562.0s        0            0.0          985.3      0.0      0.0      0.0      0.0 write
		  |   563.0s        0            0.0          983.5      0.0      0.0      0.0      0.0 write
		  |   564.0s        0            0.0          981.8      0.0      0.0      0.0      0.0 write
		  |   565.0s        0            0.0          980.1      0.0      0.0      0.0      0.0 write
		  |   566.0s        0            0.0          978.3      0.0      0.0      0.0      0.0 write
		  |   567.0s        0            0.0          976.6      0.0      0.0      0.0      0.0 write
		  |   568.0s        0            0.0          974.9      0.0      0.0      0.0      0.0 write
		  |   569.0s        0            0.0          973.2      0.0      0.0      0.0      0.0 write
		  |   570.0s        0            0.0          971.5      0.0      0.0      0.0      0.0 write
		  |   571.0s        0            0.0          969.8      0.0      0.0      0.0      0.0 write
		  |   572.0s        0            0.0          968.1      0.0      0.0      0.0      0.0 write
		  |   573.0s        0            0.0          966.4      0.0      0.0      0.0      0.0 write
		  |   574.0s        0            0.0          964.7      0.0      0.0      0.0      0.0 write
		  |   575.0s        0            0.0          963.0      0.0      0.0      0.0      0.0 write
		  |   576.0s        0            0.0          961.3      0.0      0.0      0.0      0.0 write
		  |   577.0s        0            0.0          959.7      0.0      0.0      0.0      0.0 write
		Wraps: (4) exit status 20
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *main.withCommandDetails (4) *exec.ExitError

	cluster.go:2800,kv.go:99,kv.go:186,test_runner.go:777: monitor failure: monitor task failed: t.Fatal() was called
		(1) attached stack trace
		  -- stack trace:
		  | main.(*monitor).WaitE
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2788
		  | main.(*monitor).Wait
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2796
		  | main.registerKV.func2
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/kv.go:99
		  | main.registerKV.func3
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/kv.go:186
		  | main.(*testRunner).runTest.func2
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:777
		Wraps: (2) monitor failure
		Wraps: (3) attached stack trace
		  -- stack trace:
		  | main.(*monitor).wait.func2
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2844
		Wraps: (4) monitor task failed
		Wraps: (5) attached stack trace
		  -- stack trace:
		  | main.init
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster.go:2758
		  | runtime.doInit
		  | 	/usr/local/go/src/runtime/proc.go:5652
		  | runtime.main
		  | 	/usr/local/go/src/runtime/proc.go:191
		  | runtime.goexit
		  | 	/usr/local/go/src/runtime/asm_amd64.s:1374
		Wraps: (6) t.Fatal() was called
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.leafError
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh kv0/enc=false/nodes=3/cpu=32/size=64kb
```
</p>
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*kv0/enc=false/nodes=3/cpu=32/size=64kb.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

----

```
n2
panic: open /mnt/data1/cockroach/090322.sst: too many open files [recovered]
	panic: open /mnt/data1/cockroach/090322.sst: too many open files [recovered]
	panic: open /mnt/data1/cockroach/090322.sst: too many open files

goroutine 929399 [running]:
panic(0x4938060, 0xc04304ba00)
	/usr/local/go/src/runtime/panic.go:1064 +0x545 fp=0xc01095b1f8 sp=0xc01095b130 pc=0x48b725
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc0010e4780, 0x5b7f960, 0xc040b27560)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:233 +0x126 fp=0xc01095b258 sp=0xc01095b1f8 pc=0x1581726
runtime.call32(0x0, 0x5402dc0, 0xc000af3848, 0x1800000018)
	/usr/local/go/src/runtime/asm_amd64.s:540 +0x3e fp=0xc01095b288 sp=0xc01095b258 pc=0x4c30be
runtime.reflectcallSave(0xc01095b3c8, 0x5402dc0, 0xc000af3848, 0xc000000018)
	/usr/local/go/src/runtime/panic.go:881 +0x58 fp=0xc01095b2b8 sp=0xc01095b288 pc=0x48b118
runtime.runOpenDeferFrame(0xc001f64900, 0xc000af3800, 0x0)
	/usr/local/go/src/runtime/panic.go:855 +0x2cd fp=0xc01095b348 sp=0xc01095b2b8 pc=0x48afcd
panic(0x4938060, 0xc04304ba00)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9 fp=0xc01095b410 sp=0xc01095b348 pc=0x48b399
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Send.func1(0xc01095f360, 0xc01095f400, 0xc00204a800, 0xc01095f3f8)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/store_send.go:100 +0x247 fp=0xc01095b438 sp=0xc01095b410 pc=0x200c6c7
runtime.call32(0x0, 0x53fe488, 0xc01095ec38, 0x2000000020)
	/usr/local/go/src/runtime/asm_amd64.s:540 +0x3e fp=0xc01095b468 sp=0xc01095b438 pc=0x4c30be
panic(0x4938060, 0xc04304ba00)
	/usr/local/go/src/runtime/panic.go:975 +0x47a fp=0xc01095b530 sp=0xc01095b468 pc=0x48b65a
github.com/cockroachdb/cockroach/pkg/storage.(*pebbleIterator).destroy(0xc00307fb28)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_iterator.go:724 +0x1a7 fp=0xc01095b648 sp=0xc01095b530 pc=0x1c84427
github.com/cockroachdb/cockroach/pkg/storage.(*pebbleBatch).Close(0xc00307fb00)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/pebble_batch.go:113 +0x6a fp=0xc01095b670 sp=0xc01095b648 pc=0x1c7b18a
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).evaluate1PC.func1(0xc01095c170, 0xc01095b900)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_write.go:491 +0x48 fp=0xc01095b688 sp=0xc01095b670 pc=0x2001d68
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).evaluate1PC(0xc003323b00, 0x5b7f960, 0xc040b275f0, 0xc00460a048, 0x8, 0xc037cab540, 0xc029b5ad80, 0x2, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_write.go:529 +0xed9 fp=0xc01095c138 sp=0xc01095b688 pc=0x1fa7919
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).evaluateWriteBatch(0xc003323b00, 0x5b7f960, 0xc040b275f0, 0xc00460a048, 0x8, 0xc037cab540, 0x167a9373b387af8c, 0x0, 0xc029b5ad80, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_write.go:425 +0x3a5 fp=0xc01095ca58 sp=0xc01095c138 pc=0x1fa66e5
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).evaluateProposal(0xc003323b00, 0x5b7f960, 0xc040b275f0, 0xc00460a048, 0x8, 0xc037cab540, 0x167a9373b387af8c, 0x0, 0xc029b5ad80, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_proposal.go:782 +0x157 fp=0xc01095d4d8 sp=0xc01095ca58 pc=0x1f6c9d7
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).requestToProposal(0xc003323b00, 0x5b7f960, 0xc040b275f0, 0xc00460a048, 0x8, 0xc037cab540, 0x167a92f181a59830, 0x0, 0x0, 0x200000002, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_proposal.go:913 +0xb6 fp=0xc01095d580 sp=0xc01095d4d8 pc=0x1f6d776
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).evalAndPropose(0xc003323b00, 0x5b7f960, 0xc040b275f0, 0xc037cab540, 0xc004112460, 0x167a92f181a59830, 0x0, 0x0, 0x200000002, 0x2, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_raft.go:85 +0x18e fp=0xc01095dd60 sp=0xc01095d580 pc=0x1f74dce
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).executeWriteBatch(0xc003323b00, 0x5b7f960, 0xc040b275f0, 0xc037cab540, 0xc004112460, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_write.go:144 +0x6eb fp=0xc01095e5b8 sp=0xc01095dd60 pc=0x1fa468b
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).executeBatchWithConcurrencyRetries(0xc003323b00, 0x5b7f960, 0xc040b275f0, 0xc037cab540, 0x53fe428, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_send.go:366 +0x336 fp=0xc01095e7c8 sp=0xc01095e5b8 pc=0x1f985d6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).sendWithRangeID(0xc003323b00, 0x5b7f960, 0xc040b275c0, 0x317, 0xc037cab540, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_send.go:98 +0x59d fp=0xc01095ea10 sp=0xc01095e7c8 pc=0x1f975bd
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).Send(...)
```

--- a/pkg/cmd/roachprod/install/cockroach.go
+++ b/pkg/cmd/roachprod/install/cockroach.go
@@ -393,7 +393,6 @@ set -euo pipefail
 
 if [[ "${1}" == "run" ]]; then
   local="{{if .Local}}true{{end}}"
-  ulimit -c unlimited
   mkdir -p {{.LogDir}}
   echo "cockroach start: $(date), logging to {{.LogDir}}" | tee -a {{.LogDir}}/{roachprod,cockroach.std{out,err}}.log
   {{.KeyCmd}}
@@ -450,8 +449,10 @@ fi
 # notification doesn't come from the main PID (which is bash).
 sudo systemd-run --unit cockroach \
   --same-dir --uid $(id -u) --gid $(id -g) \
-	--service-type=notify -p NotifyAccess=all \
-	-p MemoryMax={{.MemoryMax}} \
+  --service-type=notify -p NotifyAccess=all \
+  -p MemoryMax={{.MemoryMax}} \
+  -p LimitCORE=infinity \
+  -p LimitNOFILE=65536 \
 	bash $0 run
 EOF
 `)

